Remove AbstractHeap related changes from WasmOMGIRGenerator32_64

ioannad opened the pull request:
#### 46cca25a273887b13a9da688a6dad697a89793be
<pre>
Remove AbstractHeap related changes from WasmOMGIRGenerator32_64
<a href="https://bugs.webkit.org/show_bug.cgi?id=292673">https://bugs.webkit.org/show_bug.cgi?id=292673</a>

Reviewed by Justin Michaud.

This applies the reverting changes to WasmOMGIRGenerator32_64.cpp
which were made to WasmOMGIRGenerator.cpp with <a href="https://commits.webkit.org/286048@main">https://commits.webkit.org/286048@main</a>

* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::OMGIRGenerator::addConstant):
(JSC::Wasm::OMGIRGenerator::addExtractLane):
(JSC::Wasm::OMGIRGenerator::addReplaceLane):
(JSC::Wasm::OMGIRGenerator::addSIMDI_V):
(JSC::Wasm::OMGIRGenerator::addSIMDV_V):
(JSC::Wasm::OMGIRGenerator::addSIMDBitwiseSelect):
(JSC::Wasm::OMGIRGenerator::addSIMDRelOp):
(JSC::Wasm::OMGIRGenerator::fixupOutOfBoundsIndicesForSwizzle):
(JSC::Wasm::OMGIRGenerator::addSIMDV_VV):
(JSC::Wasm::OMGIRGenerator::addSIMDRelaxedFMA):
(JSC::Wasm::OMGIRGenerator::emitPrepareWasmOperation):
(JSC::Wasm::OMGIRGenerator::callWasmOperation):
(JSC::Wasm::OMGIRGenerator::get):
(JSC::Wasm::OMGIRGenerator::set):
(JSC::Wasm::OMGIRGenerator::fixupPointerPlusOffset):
(JSC::Wasm::OMGIRGenerator::restoreWasmContextInstance):
(JSC::Wasm::OMGIRGenerator::OMGIRGenerator):
(JSC::Wasm::OMGIRGenerator::restoreWebAssemblyGlobalState):
(JSC::Wasm::OMGIRGenerator::reloadMemoryRegistersFromInstance):
(JSC::Wasm::OMGIRGenerator::insertEntrySwitch):
(JSC::Wasm::OMGIRGenerator::insertConstants):
(JSC::Wasm::OMGIRGenerator::addLocal):
(JSC::Wasm::OMGIRGenerator::addInlinedArguments):
(JSC::Wasm::OMGIRGenerator::addArguments):
(JSC::Wasm::OMGIRGenerator::addRefIsNull):
(JSC::Wasm::OMGIRGenerator::addTableGet):
(JSC::Wasm::OMGIRGenerator::addTableSet):
(JSC::Wasm::OMGIRGenerator::addRefAsNonNull):
(JSC::Wasm::OMGIRGenerator::addTableInit):
(JSC::Wasm::OMGIRGenerator::addElemDrop):
(JSC::Wasm::OMGIRGenerator::addTableSize):
(JSC::Wasm::OMGIRGenerator::addTableGrow):
(JSC::Wasm::OMGIRGenerator::addTableFill):
(JSC::Wasm::OMGIRGenerator::addTableCopy):
(JSC::Wasm::OMGIRGenerator::getLocal):
(JSC::Wasm::OMGIRGenerator::addUnreachable):
(JSC::Wasm::OMGIRGenerator::addCrash):
(JSC::Wasm::OMGIRGenerator::emitIndirectCall):
(JSC::Wasm::OMGIRGenerator::addCurrentMemory):
(JSC::Wasm::OMGIRGenerator::addMemoryFill):
(JSC::Wasm::OMGIRGenerator::addMemoryInit):
(JSC::Wasm::OMGIRGenerator::addMemoryCopy):
(JSC::Wasm::OMGIRGenerator::addDataDrop):
(JSC::Wasm::OMGIRGenerator::setLocal):
(JSC::Wasm::OMGIRGenerator::teeLocal):
(JSC::Wasm::OMGIRGenerator::getGlobal):
(JSC::Wasm::OMGIRGenerator::setGlobal):
(JSC::Wasm::OMGIRGenerator::emitWriteBarrier):
(JSC::Wasm::OMGIRGenerator::emitCheckAndPreparePointer):
(JSC::Wasm::OMGIRGenerator::emitLoadOp):
(JSC::Wasm::OMGIRGenerator::load):
(JSC::Wasm::OMGIRGenerator::emitStoreOp):
(JSC::Wasm::OMGIRGenerator::store):
(JSC::Wasm::OMGIRGenerator::sanitizeAtomicResult):
(JSC::Wasm::OMGIRGenerator::fixupPointerPlusOffsetForAtomicOps):
(JSC::Wasm::OMGIRGenerator::emitAtomicLoadOp):
(JSC::Wasm::OMGIRGenerator::atomicLoad):
(JSC::Wasm::OMGIRGenerator::emitAtomicStoreOp):
(JSC::Wasm::OMGIRGenerator::atomicStore):
(JSC::Wasm::OMGIRGenerator::emitAtomicBinaryRMWOp):
(JSC::Wasm::OMGIRGenerator::atomicBinaryRMW):
(JSC::Wasm::OMGIRGenerator::emitAtomicCompareExchange):
(JSC::Wasm::OMGIRGenerator::emitStructSet):
(JSC::Wasm::OMGIRGenerator::atomicCompareExchange):
(JSC::Wasm::OMGIRGenerator::atomicWait):
(JSC::Wasm::OMGIRGenerator::atomicNotify):
(JSC::Wasm::OMGIRGenerator::atomicFence):
(JSC::Wasm::OMGIRGenerator::truncSaturated):
(JSC::Wasm::OMGIRGenerator::addI31GetS):
(JSC::Wasm::OMGIRGenerator::addI31GetU):
(JSC::Wasm::OMGIRGenerator::pushArrayNew):
(JSC::Wasm::OMGIRGenerator::addArrayNew):
(JSC::Wasm::OMGIRGenerator::pushArrayNewFromSegment):
(JSC::Wasm::OMGIRGenerator::addArrayNewDefault):
(JSC::Wasm::OMGIRGenerator::addArrayNewFixed):
(JSC::Wasm::OMGIRGenerator::addArrayGet):
(JSC::Wasm::OMGIRGenerator::emitArrayNullCheck):
(JSC::Wasm::OMGIRGenerator::emitArraySetUnchecked):
(JSC::Wasm::OMGIRGenerator::addArraySet):
(JSC::Wasm::OMGIRGenerator::addArrayLen):
(JSC::Wasm::OMGIRGenerator::AbstractHeap::AbstractHeap): Deleted.
(JSC::Wasm::OMGIRGenerator::AbstractHeap::addChild): Deleted.
(JSC::Wasm::OMGIRGenerator::AbstractHeapRange::AbstractHeapRange): Deleted.
(JSC::Wasm::OMGIRGenerator::initializeHeaps): Deleted.
(JSC::Wasm::OMGIRGenerator::recordHeap): Deleted.
(JSC::Wasm::OMGIRGenerator::append): Deleted.
(JSC::Wasm::OMGIRGenerator::finalizeHeapRanges): Deleted.
(JSC::Wasm::OMGIRGenerator::computeHeapRanges): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::HeapRecorder&lt;MemoryValue&gt;::recordHeap): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::HeapRecorder&lt;AtomicValue&gt;::recordHeap): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender::append): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender&lt;MemoryValue&gt;::append): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender&lt;AtomicValue&gt;::append): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayFill): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayCopy): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayInitElem): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayInitData): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructNew): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructNewDefault): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructGet): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructSet): Deleted.
(JSC::Wasm::OMGIRGenerator::addRefTest): Deleted.
(JSC::Wasm::OMGIRGenerator::addRefCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitRefTestOrCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCheckOrBranchForCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitLoadRTTFromFuncref): Deleted.
(JSC::Wasm::OMGIRGenerator::decodeNonNullStructure): Deleted.
(JSC::Wasm::OMGIRGenerator::emitLoadRTTFromObject): Deleted.
(JSC::Wasm::OMGIRGenerator::emitNotRTTKind): Deleted.
(JSC::Wasm::OMGIRGenerator::addAnyConvertExtern): Deleted.
(JSC::Wasm::OMGIRGenerator::addExternConvertAny): Deleted.
(JSC::Wasm::OMGIRGenerator::addSelect): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDSplat): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDShift): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDExtmul): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDShuffle): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoad): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDStore): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadSplat): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadLane): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDStoreLane): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadExtend): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadPad): Deleted.
(JSC::Wasm::OMGIRGenerator::loadFromScratchBuffer): Deleted.
(JSC::Wasm::OMGIRGenerator::connectControlAtEntrypoint): Deleted.
(JSC::Wasm::OMGIRGenerator::addLoop): Deleted.
(JSC::Wasm::OMGIRGenerator::addTopLevel): Deleted.
(JSC::Wasm::OMGIRGenerator::addBlock): Deleted.
(JSC::Wasm::OMGIRGenerator::addIf): Deleted.
(JSC::Wasm::OMGIRGenerator::addElse): Deleted.
(JSC::Wasm::OMGIRGenerator::addElseToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addTry): Deleted.
(JSC::Wasm::OMGIRGenerator::addTryTable): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatch): Deleted.
(JSC::Wasm::OMGIRGenerator::preparePatchpointForExceptions): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchAll): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchAllToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCatchImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCatchTableImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::addDelegate): Deleted.
(JSC::Wasm::OMGIRGenerator::addDelegateToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addThrow): Deleted.
(JSC::Wasm::OMGIRGenerator::addThrowRef): Deleted.
(JSC::Wasm::OMGIRGenerator::addRethrow): Deleted.
(JSC::Wasm::OMGIRGenerator::addInlinedReturn): Deleted.
(JSC::Wasm::OMGIRGenerator::addReturn): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranch): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranchNull): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranchCast): Deleted.
(JSC::Wasm::OMGIRGenerator::addSwitch): Deleted.
(JSC::Wasm::OMGIRGenerator::endBlock): Deleted.
(JSC::Wasm::OMGIRGenerator::createCallConstrainedArgs): Deleted.
(JSC::Wasm::OMGIRGenerator::createCallPatchpoint): Deleted.
(JSC::Wasm::prepareForTailCallImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::createTailCallPatchpoint): Deleted.
(JSC::Wasm::OMGIRGenerator::canInline const): Deleted.
(JSC::Wasm::OMGIRGenerator::emitInlineDirectCall): Deleted.
(JSC::Wasm::OMGIRGenerator::addCall): Deleted.
(JSC::Wasm::OMGIRGenerator::addCallIndirect): Deleted.
(JSC::Wasm::OMGIRGenerator::addCallRef): Deleted.
(JSC::Wasm::OMGIRGenerator::unify): Deleted.
(JSC::Wasm::OMGIRGenerator::unifyValuesWithBlock): Deleted.
(JSC::Wasm::dumpExpressionStack): Deleted.
(JSC::Wasm::OMGIRGenerator::dump): Deleted.
(JSC::Wasm::OMGIRGenerator::origin): Deleted.
(JSC::Wasm::shouldDumpIRFor): Deleted.
(JSC::Wasm::parseAndCompileOMG): Deleted.
(JSC::Wasm::OMGIRGenerator::emitChecksForModOrDiv): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32DivS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32RemS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32DivU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32RemU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64DivS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64RemS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64DivU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64RemU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32Ctz): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64Ctz): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32Popcnt): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64Popcnt): Deleted.
(JSC::Wasm::OMGIRGenerator::addF64ConvertUI64): Deleted.
(JSC::Wasm::OMGIRGenerator::addF32ConvertUI64): Deleted.
(JSC::Wasm::OMGIRGenerator::addF64Nearest): Deleted.
(JSC::Wasm::OMGIRGenerator::addF32Nearest): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncSF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncSF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncUF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncUF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncSF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncUF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncSF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncUF32): Deleted.

Canonical link: <a href="https://commits.webkit.org/294684@main">https://commits.webkit.org/294684@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4c01c48fee722cb4739cfa3666896eeef53dcf2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/102616 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/22285 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/12599 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/107777 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/53253 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/22593 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/30791 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/78055 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/35034 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/105622 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/17505 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/92612 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/58390 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/17344 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/10644 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/52610 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/95287 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/87155 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/10712 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/110152 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/101225 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/29748 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21931 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/87034 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/30112 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/88809 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/86641 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22067 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/31467 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/9194 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/23999 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/29676 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/34988 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/124859 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/29487 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34654 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/32810 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/31046 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aoikonomopoulos commented on Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
```diff
@@ -98,6 +98,7 @@ using ConstIntPtrValue = Const32Value;
 using ConstIntPtrValue = Const64Value;
 #endif
 
+namespace {
```
Why is this extra namespace needed here?

ioannad replied:
This is there to keep this file as close as possible to the 64bit version. In particular, the [reverting changes on WasmOMGIRGenerator.cpp also re-add this extra namespace](https://github.com/WebKit/WebKit/pull/36068/files#diff-e0cc9beb26b9d2cbf93a0dbc593d1ccfbd186ade0356b40ff01a87a85e5c0486R98).

ioannad committed:
Remove AbstractHeap related changes from WasmOMGIRGenerator32_64
https://bugs.webkit.org/show_bug.cgi?id=292673

Reviewed by Justin Michaud.

This applies the reverting changes to WasmOMGIRGenerator32_64.cpp
which were made to WasmOMGIRGenerator.cpp with https://commits.webkit.org/286048@main

* Source/JavaScriptCore/wasm/WasmOMGIRGenerator32_64.cpp:
(JSC::Wasm::OMGIRGenerator::addConstant):
(JSC::Wasm::OMGIRGenerator::addExtractLane):
(JSC::Wasm::OMGIRGenerator::addReplaceLane):
(JSC::Wasm::OMGIRGenerator::addSIMDI_V):
(JSC::Wasm::OMGIRGenerator::addSIMDV_V):
(JSC::Wasm::OMGIRGenerator::addSIMDBitwiseSelect):
(JSC::Wasm::OMGIRGenerator::addSIMDRelOp):
(JSC::Wasm::OMGIRGenerator::fixupOutOfBoundsIndicesForSwizzle):
(JSC::Wasm::OMGIRGenerator::addSIMDV_VV):
(JSC::Wasm::OMGIRGenerator::addSIMDRelaxedFMA):
(JSC::Wasm::OMGIRGenerator::emitPrepareWasmOperation):
(JSC::Wasm::OMGIRGenerator::callWasmOperation):
(JSC::Wasm::OMGIRGenerator::get):
(JSC::Wasm::OMGIRGenerator::set):
(JSC::Wasm::OMGIRGenerator::fixupPointerPlusOffset):
(JSC::Wasm::OMGIRGenerator::restoreWasmContextInstance):
(JSC::Wasm::OMGIRGenerator::OMGIRGenerator):
(JSC::Wasm::OMGIRGenerator::restoreWebAssemblyGlobalState):
(JSC::Wasm::OMGIRGenerator::reloadMemoryRegistersFromInstance):
(JSC::Wasm::OMGIRGenerator::insertEntrySwitch):
(JSC::Wasm::OMGIRGenerator::insertConstants):
(JSC::Wasm::OMGIRGenerator::addLocal):
(JSC::Wasm::OMGIRGenerator::addInlinedArguments):
(JSC::Wasm::OMGIRGenerator::addArguments):
(JSC::Wasm::OMGIRGenerator::addRefIsNull):
(JSC::Wasm::OMGIRGenerator::addTableGet):
(JSC::Wasm::OMGIRGenerator::addTableSet):
(JSC::Wasm::OMGIRGenerator::addRefAsNonNull):
(JSC::Wasm::OMGIRGenerator::addTableInit):
(JSC::Wasm::OMGIRGenerator::addElemDrop):
(JSC::Wasm::OMGIRGenerator::addTableSize):
(JSC::Wasm::OMGIRGenerator::addTableGrow):
(JSC::Wasm::OMGIRGenerator::addTableFill):
(JSC::Wasm::OMGIRGenerator::addTableCopy):
(JSC::Wasm::OMGIRGenerator::getLocal):
(JSC::Wasm::OMGIRGenerator::addUnreachable):
(JSC::Wasm::OMGIRGenerator::addCrash):
(JSC::Wasm::OMGIRGenerator::emitIndirectCall):
(JSC::Wasm::OMGIRGenerator::addCurrentMemory):
(JSC::Wasm::OMGIRGenerator::addMemoryFill):
(JSC::Wasm::OMGIRGenerator::addMemoryInit):
(JSC::Wasm::OMGIRGenerator::addMemoryCopy):
(JSC::Wasm::OMGIRGenerator::addDataDrop):
(JSC::Wasm::OMGIRGenerator::setLocal):
(JSC::Wasm::OMGIRGenerator::teeLocal):
(JSC::Wasm::OMGIRGenerator::getGlobal):
(JSC::Wasm::OMGIRGenerator::setGlobal):
(JSC::Wasm::OMGIRGenerator::emitWriteBarrier):
(JSC::Wasm::OMGIRGenerator::emitCheckAndPreparePointer):
(JSC::Wasm::OMGIRGenerator::emitLoadOp):
(JSC::Wasm::OMGIRGenerator::load):
(JSC::Wasm::OMGIRGenerator::emitStoreOp):
(JSC::Wasm::OMGIRGenerator::store):
(JSC::Wasm::OMGIRGenerator::sanitizeAtomicResult):
(JSC::Wasm::OMGIRGenerator::fixupPointerPlusOffsetForAtomicOps):
(JSC::Wasm::OMGIRGenerator::emitAtomicLoadOp):
(JSC::Wasm::OMGIRGenerator::atomicLoad):
(JSC::Wasm::OMGIRGenerator::emitAtomicStoreOp):
(JSC::Wasm::OMGIRGenerator::atomicStore):
(JSC::Wasm::OMGIRGenerator::emitAtomicBinaryRMWOp):
(JSC::Wasm::OMGIRGenerator::atomicBinaryRMW):
(JSC::Wasm::OMGIRGenerator::emitAtomicCompareExchange):
(JSC::Wasm::OMGIRGenerator::emitStructSet):
(JSC::Wasm::OMGIRGenerator::atomicCompareExchange):
(JSC::Wasm::OMGIRGenerator::atomicWait):
(JSC::Wasm::OMGIRGenerator::atomicNotify):
(JSC::Wasm::OMGIRGenerator::atomicFence):
(JSC::Wasm::OMGIRGenerator::truncSaturated):
(JSC::Wasm::OMGIRGenerator::addI31GetS):
(JSC::Wasm::OMGIRGenerator::addI31GetU):
(JSC::Wasm::OMGIRGenerator::pushArrayNew):
(JSC::Wasm::OMGIRGenerator::addArrayNew):
(JSC::Wasm::OMGIRGenerator::pushArrayNewFromSegment):
(JSC::Wasm::OMGIRGenerator::addArrayNewDefault):
(JSC::Wasm::OMGIRGenerator::addArrayNewFixed):
(JSC::Wasm::OMGIRGenerator::addArrayGet):
(JSC::Wasm::OMGIRGenerator::emitArrayNullCheck):
(JSC::Wasm::OMGIRGenerator::emitArraySetUnchecked):
(JSC::Wasm::OMGIRGenerator::addArraySet):
(JSC::Wasm::OMGIRGenerator::addArrayLen):
(JSC::Wasm::OMGIRGenerator::AbstractHeap::AbstractHeap): Deleted.
(JSC::Wasm::OMGIRGenerator::AbstractHeap::addChild): Deleted.
(JSC::Wasm::OMGIRGenerator::AbstractHeapRange::AbstractHeapRange): Deleted.
(JSC::Wasm::OMGIRGenerator::initializeHeaps): Deleted.
(JSC::Wasm::OMGIRGenerator::recordHeap): Deleted.
(JSC::Wasm::OMGIRGenerator::append): Deleted.
(JSC::Wasm::OMGIRGenerator::finalizeHeapRanges): Deleted.
(JSC::Wasm::OMGIRGenerator::computeHeapRanges): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::HeapRecorder<MemoryValue>::recordHeap): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::HeapRecorder<AtomicValue>::recordHeap): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender::append): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender<MemoryValue>::append): Deleted.
(JSC::Wasm::WasmOMGIRGeneratorInternal::ValueAppender<AtomicValue>::append): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayFill): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayCopy): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayInitElem): Deleted.
(JSC::Wasm::OMGIRGenerator::addArrayInitData): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructNew): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructNewDefault): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructGet): Deleted.
(JSC::Wasm::OMGIRGenerator::addStructSet): Deleted.
(JSC::Wasm::OMGIRGenerator::addRefTest): Deleted.
(JSC::Wasm::OMGIRGenerator::addRefCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitRefTestOrCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCheckOrBranchForCast): Deleted.
(JSC::Wasm::OMGIRGenerator::emitLoadRTTFromFuncref): Deleted.
(JSC::Wasm::OMGIRGenerator::decodeNonNullStructure): Deleted.
(JSC::Wasm::OMGIRGenerator::emitLoadRTTFromObject): Deleted.
(JSC::Wasm::OMGIRGenerator::emitNotRTTKind): Deleted.
(JSC::Wasm::OMGIRGenerator::addAnyConvertExtern): Deleted.
(JSC::Wasm::OMGIRGenerator::addExternConvertAny): Deleted.
(JSC::Wasm::OMGIRGenerator::addSelect): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDSplat): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDShift): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDExtmul): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDShuffle): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoad): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDStore): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadSplat): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadLane): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDStoreLane): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadExtend): Deleted.
(JSC::Wasm::OMGIRGenerator::addSIMDLoadPad): Deleted.
(JSC::Wasm::OMGIRGenerator::loadFromScratchBuffer): Deleted.
(JSC::Wasm::OMGIRGenerator::connectControlAtEntrypoint): Deleted.
(JSC::Wasm::OMGIRGenerator::addLoop): Deleted.
(JSC::Wasm::OMGIRGenerator::addTopLevel): Deleted.
(JSC::Wasm::OMGIRGenerator::addBlock): Deleted.
(JSC::Wasm::OMGIRGenerator::addIf): Deleted.
(JSC::Wasm::OMGIRGenerator::addElse): Deleted.
(JSC::Wasm::OMGIRGenerator::addElseToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addTry): Deleted.
(JSC::Wasm::OMGIRGenerator::addTryTable): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatch): Deleted.
(JSC::Wasm::OMGIRGenerator::preparePatchpointForExceptions): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchAll): Deleted.
(JSC::Wasm::OMGIRGenerator::addCatchAllToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCatchImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::emitCatchTableImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::addDelegate): Deleted.
(JSC::Wasm::OMGIRGenerator::addDelegateToUnreachable): Deleted.
(JSC::Wasm::OMGIRGenerator::addThrow): Deleted.
(JSC::Wasm::OMGIRGenerator::addThrowRef): Deleted.
(JSC::Wasm::OMGIRGenerator::addRethrow): Deleted.
(JSC::Wasm::OMGIRGenerator::addInlinedReturn): Deleted.
(JSC::Wasm::OMGIRGenerator::addReturn): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranch): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranchNull): Deleted.
(JSC::Wasm::OMGIRGenerator::addBranchCast): Deleted.
(JSC::Wasm::OMGIRGenerator::addSwitch): Deleted.
(JSC::Wasm::OMGIRGenerator::endBlock): Deleted.
(JSC::Wasm::OMGIRGenerator::createCallConstrainedArgs): Deleted.
(JSC::Wasm::OMGIRGenerator::createCallPatchpoint): Deleted.
(JSC::Wasm::prepareForTailCallImpl): Deleted.
(JSC::Wasm::OMGIRGenerator::createTailCallPatchpoint): Deleted.
(JSC::Wasm::OMGIRGenerator::canInline const): Deleted.
(JSC::Wasm::OMGIRGenerator::emitInlineDirectCall): Deleted.
(JSC::Wasm::OMGIRGenerator::addCall): Deleted.
(JSC::Wasm::OMGIRGenerator::addCallIndirect): Deleted.
(JSC::Wasm::OMGIRGenerator::addCallRef): Deleted.
(JSC::Wasm::OMGIRGenerator::unify): Deleted.
(JSC::Wasm::OMGIRGenerator::unifyValuesWithBlock): Deleted.
(JSC::Wasm::dumpExpressionStack): Deleted.
(JSC::Wasm::OMGIRGenerator::dump): Deleted.
(JSC::Wasm::OMGIRGenerator::origin): Deleted.
(JSC::Wasm::shouldDumpIRFor): Deleted.
(JSC::Wasm::parseAndCompileOMG): Deleted.
(JSC::Wasm::OMGIRGenerator::emitChecksForModOrDiv): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32DivS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32RemS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32DivU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32RemU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64DivS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64RemS): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64DivU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64RemU): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32Ctz): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64Ctz): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32Popcnt): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64Popcnt): Deleted.
(JSC::Wasm::OMGIRGenerator::addF64ConvertUI64): Deleted.
(JSC::Wasm::OMGIRGenerator::addF32ConvertUI64): Deleted.
(JSC::Wasm::OMGIRGenerator::addF64Nearest): Deleted.
(JSC::Wasm::OMGIRGenerator::addF32Nearest): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncSF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncSF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncUF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI32TruncUF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncSF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncUF64): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncSF32): Deleted.
(JSC::Wasm::OMGIRGenerator::addI64TruncUF32): Deleted.

Canonical link: https://commits.webkit.org/294684@main
